### PR TITLE
Install java via native tools in CI

### PR DESCRIPTION
**Pull request by mmitche**
The VMR doesn't have Java installed globally on the machines, but it does have it in the native tool cache like other CI machines. This PR updates the build scripts to use the native tool cache to install Java in CI builds. I've also updated to 11.0.22 (vs. the .3 version from 2019) and the script used to pull this from native tool assets storage account.

### CODE REVIEW

**Comment by mmitche** [comment]
Official test: https://dev.azure.com/dnceng/internal/_build/results?buildId=2397276&view=results

File: global.json [hunk 1]
```diff
@@ -12,8 +12,6 @@
         "$(MicrosoftNETCoreBrowserDebugHostTransportVersion)"
       ]
     },
-    "Git": "2.22.0",
```
Comment (wtgodbe) [inline global.json:15]: Why remove git?

**Review by wtgodbe — APPROVED** [review]

File: global.json (hunk 1 above)
Comment (mmitche) [reply global.json:15]: This doesn't do anything :)

**Comment by mmitche** [comment]
This needs a tweak or two.

**Comment by amcasey** [comment]
Do we need something similar for [wasm-tools](https://dev.azure.com/dnceng-public/public/_build/results?buildId=594188&view=logs&j=fe94c0c9-bb8c-5d6f-3b51-887173cc2f5c&t=7877c2d9-a5b1-59a3-dc2b-0d7e9d9db055&l=69)?

**Comment by mmitche** [comment]
> This needs a tweak or two.

Because the native tools are laid out in a weird way...this doenst' work locally...i'll need to disable native tools installs unless the native tools are on the machine

**Comment by MackinnonBuck** [comment]
@mmitche, we've been seeing a lot of failures in `main` since this merged:
```text
Could not find the JDK. Either run D:\a\_work\1\s\eng\scripts\InstallJdk.ps1 to install for this repo, or install the JDK globally on your machine (see D:\a\_work\1\s\eng\..\docs\BuildFromSource.md for details).
```
Do we need to revert this? Or do a follow-up?

**Pull request merged**